internal/dag/gatewayapi_processor.go

```diff
@@ -881,39 +882,40 @@ func (p *GatewayAPIProcessor) computeGatewayConditions(gwAccessor *status.Gatewa
 	)
 
 	switch {
-	case gatewayNotReadyCondition != nil:
+	case gatewayNotProgrammedCondition != nil:
 		gwAccessor.AddCondition(
-			gatewayapi_v1beta1.GatewayConditionType(gatewayNotReadyCondition.Type),
-			gatewayNotReadyCondition.Status,
-			gatewayapi_v1beta1.GatewayConditionReason(gatewayNotReadyCondition.Reason),
-			gatewayNotReadyCondition.Message,
+			gatewayapi_v1beta1.GatewayConditionType(gatewayNotProgrammedCondition.Type),
+			gatewayNotProgrammedCondition.Status,
+			gatewayapi_v1beta1.GatewayConditionReason(gatewayNotProgrammedCondition.Reason),
+			gatewayNotProgrammedCondition.Message,
 		)
 	default:
-		// Check for any listeners with a Ready: false condition.
-		allListenersReady := true
+		// Check for any listeners with a Programmed: false condition.
+		allListenersProgrammed := true
 		for _, ls := range gwAccessor.ListenerStatus {
 			if ls == nil {
 				continue
 			}
 
 			for _, cond := range ls.Conditions {
-				if cond.Type == string(gatewayapi_v1beta1.ListenerConditionReady) && cond.Status == metav1.ConditionFalse {
-					allListenersReady = false
+				if cond.Type == string(gatewayapi_v1beta1.ListenerConditionProgrammed) && cond.Status == metav1.ConditionFalse {
+					allListenersProgrammed = false
 					break
 				}
 			}
 
-			if !allListenersReady {
+			if !allListenersProgrammed {
 				break
 			}
 		}
 
-		if !allListenersReady {
-			// If we have invalid listeners, set Ready=false.
-			gwAccessor.AddCondition(gatewayapi_v1beta1.GatewayConditionReady, metav1.ConditionFalse, gatewayapi_v1beta1.GatewayReasonListenersNotValid, "Listeners are not valid")
+		if !allListenersProgrammed {
+			// If we have invalid listeners, set Programmed=false.
+			// TODO(sk) resolve condition type-reason mismatch
```

Comment:
Similarly here, the "ListenersNotValid" reason did not move over to the "Programmed" condition (ref. https://github.com/kubernetes-sigs/gateway-api/blob/eb822e981bd8f7b4f52e89cafbe2d0397b5f01fb/apis/v1beta1/gateway_types.go#L634-L637), so need to decide what to do here.

Reply:
yeah the wording on that Reason makes it sounds like it should be used with the Programmed Condition

`one or more Listeners have an invalid or unsupported configuration and cannot be configured on the Gateway.`

good for us to keep now and open an upstream issue

Reply:
Agreed. Will raise an upstream issue.